src/helpers/ipfs.ts

```diff
@@ -113,6 +119,55 @@ export class TernoaIPFS {
     return await TernoaIPFS.storeFile(service, metadataFile)
   }
 
+  /**
+   * Store a Ternoa secret NFT's metadata & asset on IPFS.
+   *
+   * @param service
+   * @param encryptedFile     NFT's encrypted asset.
+   * @param publicKey         Public key used to encrypt the Secret NFT.
+   * @param metadata          Ternoa secret NFT metadata structure {@link https://github.com/capsule-corp-ternoa/ternoa-proposals/blob/main/TIPs/tip-510-Secret-nft.md here}.
+   * @param encryptedFileType The original encrypted file type.
+   * @returns                 IPFS data (Hash, Size, Name).
+   */
+  static storeSecretNFT = async <T>(
+    service: IServiceIPFS,
+    encryptedFile: string,
+    publicKey?: string,
+    metadata?: SecretNftMetadataType<T>,
+    encryptedFileType?: string,
+  ) => {
+    if (metadata) validateSecretNFTMetadata(metadata)
+    let publicNFTKeyHash
+    if (publicKey) {
+      if (typeof publicKey !== "string")
+        throw new TypeError(`${Errors.IPFS_METADATA_VALIDATION_ERROR} : Secret NFT's publicKey must be a string`)
+      const publicKeyBlob = new Blob([publicKey], { type: "text/plain" })
+      const publicKeyFile = new File([publicKeyBlob], "SecretNFT public key")
+      const publicKeyRes = await TernoaIPFS.storeFile(service, publicKeyFile)
+      if (!publicKeyRes) throw new Error(`${Errors.IPFS_FILE_UPLOAD_ERROR} - Unable to upload secret NFT's public key`)
+      publicNFTKeyHash = publicKeyRes.Hash
+    }
+    const blob = new Blob([encryptedFile], { type: "text/plain" })
+    const file = new File([blob], "SecretNFT metadata")
+    const secretNFTRes = await TernoaIPFS.storeFile(service, file)
+    if (!secretNFTRes) throw new Error(`${Errors.IPFS_FILE_UPLOAD_ERROR} - Unable to upload secret NFT's asset`)
+    const { Hash: secretNFTHash, Size: secretNFTSize } = secretNFTRes
+    const secretNFTMetadata = {
+      ...(metadata && metadata),
+      properties: {
+        encryptedMedia: {
+          hash: secretNFTHash,
+          type: encryptedFileType ?? file.type,
+          size: secretNFTSize,
+        },
+        nftPublicKey: publicNFTKeyHash && publicNFTKeyHash,
```

Comment:
Metadatas can be written it like this below since `nftPublicKey` is optional according [TIP 510](https://github.com/capsule-corp-ternoa/ternoa-proposals/blob/main/TIPs/tip-510-Secret-nft.md#metadata):

```typescript
const secretNFTMetadata = {
      ...(metadata && metadata),
      properties: {
        encryptedMedia: {
          hash: secretNFTHash,
          type: encryptedFileType ?? file.type,
          size: secretNFTSize,
        },
        ...(nftPublicKeyHash && { nftPublicKey: nftPublicKeyHash }),
      },
    }

```

Reply:
Cleaner this way indeed! :) nftPublicKey was not added to properties when no publicNFTKeyHash was provided

Reply:
ok cool :)